api/app/signals/apps/email_integrations/tasks.py

```diff
@@ -1,9 +1,70 @@
 # SPDX-License-Identifier: MPL-2.0
 # Copyright (C) 2021 - 2022 Gemeente Amsterdam
+import logging
+
+from django.contrib.auth import get_user_model
+
 from signals.apps.email_integrations.services import MailService
+from signals.apps.signals.models import Department, Signal
 from signals.celery import app
 
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
 
 @app.task
 def send_mail_reporter(pk):
     MailService.status_mail(signal=pk)
+
+
+@app.task
+def send_mail_assigned_signal_departments(signal_pk, departments_pk):
+    try:
+        signal = Signal.objects.get(pk=signal_pk)
+    except Signal.DoesNotExist:
+        logger.exception()
+        return
+
+    for department_pk in departments_pk:
+        try:
+            department = Department.objects.get(pk=department_pk)
+        except Department.DoesNotExist:
+            logger.exception()
+            continue
+
+        user_profiles = department.user_profiles.select_related('user').filter(
+            user__is_active=True,
+            notification_on_department_assignment=True
+        )
+
+        for profile in user_profiles:
+            MailService.system_mail(signal=signal,
```

Comment:
Small comment: this call would remain the same if you switch your `AssignedAction` to using the `_required_call_kwargs` way of calling it. Same for the call below.

Reply:
Now using `required_call_kwargs`